### PR TITLE
[CSYS-318] Migrating Tag components

### PR DESCRIPTION
**Pull request by tiidadavena**
Link to task: [Card from Jira](https://labcodes.atlassian.net/browse/CSYS-318)

Staging app:
https://CSYS-318-migrate-tag-components--confetti-storybook.netlify.app/

How to test:
Run the project locally and check if the Tag's subcomponents are working as expected.

Description of your solution:
Removing Class Components' methods and using Hooks to control the states and other properties.

Screenshots (when applicable):
N/A

### CODE REVIEW

File: src/Dropdown/TagDropdownTrigger.js [hunk 1]
```diff
@@ -1,48 +1,97 @@
...
+        className="lab-tag--dropdown-icon"
       />
-    );
-  }
+    </span>
+  );
+
+  return (
+    <AbstractTag
+      className={`lab-tag--dropdown${`${
```
Comment (lucianoratamero) [inline src/Dropdown/TagDropdownTrigger.js:43]: erhm, I think this template string is a bit messed up 😅

File: src/Tags/AbstractTag.js [hunk 2]
```diff
@@ -1,151 +1,145 @@
...
+  return (
+    <span
+      className={
+        `lab-tag ${className}` +
+        `${disabled || ariaDisabled ? ` lab-tag--disabled` : ""}` +
+        `${isOutline ? ` lab-tag--outline` : ""}` +
+        `${color ? ` lab-tag--${color}-${skin}` : ` lab-tag--${skin}`}`
+      }
+      onClick={!(ariaDisabled || disabled) ? handleEvent : () => {}}
+      onKeyDown={handleEvent}
```
[the remark this answers is not in the document]
Comment (lucianoratamero) [reply src/Tags/AbstractTag.js:80]: apparently, it's deprecated: https://www.delftstack.com/pt/howto/react/onkeypress-react/#onkeypress-em-react

[the remark this answers is not in the document]
Comment (r-lelis) [reply src/Tags/AbstractTag.js:80]: Oh, okay!

File: src/Dropdown/TagDropdownTrigger.js [hunk 3]
```diff
@@ -1,48 +1,97 @@
...
+        icon ? ` lab-tag--has-left-icon` : ""
+      }`}`}
```
Comment (r-lelis) [inline src/Dropdown/TagDropdownTrigger.js:45]: > erhm, I think this template string is a bit messed up sweat_smile

Yeah, I think they are =]
We could do something like this:
```suggestion
      className={`lab-tag--dropdown${icon ? ` lab-tag--has-left-icon` : ""}`}
```

**Review by r-lelis — APPROVED** [review]
LGTM!

**Pull request merged**